File: test/cpp/polyglot/transform/map.cpp
// Copyright 2015 Adam Grandquist

#include "./catch.hpp"
#include "./ReQL.hpp"

using namespace ReQL;

TEST_CASE("cpp Tests the RQL `map` function", "[cpp][ast]") {

  SECTION("test0") {
    std::string src0("STREAM", 6);
    Query var0(src0);
  }

  SECTION("test1") {
    std::string src0("STREAM", 6);
    Query var0(src0);
  }

  SECTION("test2") {
    std::string src0("ARRAY", 5);
    Query var0(src0);
  }

  SECTION("test3") {
    std::vector<Query> arr0(3);

    double num1(0);
    Query var1(num1);

    arr0.insert(arr0.end(), var1);

    double num2(0);
    Query var2(num2);

    arr0.insert(arr0.end(), var2);

    double num3(0);
    Query var3(num3);

    arr0.insert(arr0.end(), var3);

    Query var0(arr0);
  }

  SECTION("test4") {
    std::vector<Query> arr0(3);

    double num1(0);
    Query var1(num1);

    arr0.insert(arr0.end(), var1);

    double num2(0);
    Query var2(num2);

    arr0.insert(arr0.end(), var2);

    double num3(0);
    Query var3(num3);

    arr0.insert(arr0.end(), var3);

    Query var0(arr0);
  }

  SECTION("test5") {
    std::vector<Query> arr0(1);

    std::vector<Query> arr1(1);

    double num2(1);
    Query var2(num2);

    arr1.insert(arr1.end(), var2);

    Query var1(arr1);

    arr0.insert(arr0.end(), var1);

    Query var0(arr0);
  }

  SECTION("test6") {
    std::vector<Query> arr0(1);

    std::vector<Query> arr1(2);

    double num2(1);
    Query var2(num2);

    arr1.insert(arr1.end(), var2);

    double num3(1);
    Query var3(num3);

    arr1.insert(arr1.end(), var3);

    Query var1(arr1);

    arr0.insert(arr0.end(), var1);

    Query var0(arr0);
  }

  SECTION("test7") {
    std::vector<Query> arr0(1);

    std::vector<Query> arr1(3);

    double num2(1);
    Query var2(num2);

    arr1.insert(arr1.end(), var2);

    double num3(1);
    Query var3(num3);

    arr1.insert(arr1.end(), var3);

    double num4(1);
    Query var4(num4);

    arr1.insert(arr1.end(), var4);

    Query var1(arr1);

    arr0.insert(arr0.end(), var1);

    Query var0(arr0);
  }

  SECTION("test8") {
    std::map<std::string, Query> map0;

    std::string src1("func", 4);
    Query var1(src1);

    std::string src2("err", 3);
    Query var2(src2);

    map0.insert({src1, var2});

    Query var0(map0);
  }

  SECTION("test9") {
    std::map<std::string, Query> map0;

    std::string src1("func", 4);
    Query var1(src1);

    std::string src2("err", 3);
    Query var2(src2);

    map0.insert({src1, var2});

    Query var0(map0);
  }

  SECTION("test10") {
    std::vector<Query> arr0;
    Query var0(arr0);
  }

  SECTION("test11") {
    std::vector<Query> arr0(2);

    std::vector<Query> arr1(2);

    double num2(1);
    Query var2(num2);

    arr1.insert(arr1.end(), var2);

    double num3(1);
    Query var3(num3);

    arr1.insert(arr1.end(), var3);

    Query var1(arr1);

    arr0.insert(arr0.end(), var1);

    std::vector<Query> arr4(2);

    double num5(2);
    Query var5(num5);

    arr4.insert(arr4.end(), var5);

    double num6(2);
    Query var6(num6);

    arr4.insert(arr4.end(), var6);

    Query var4(arr4);

    arr0.insert(arr0.end(), var4);

    Query var0(arr0);
  }

  SECTION("test12") {
    std::vector<Query> arr0(2);

    std::vector<Query> arr1(2);

    double num2(0);
    Query var2(num2);

    arr1.insert(arr1.end(), var2);

    double num3(0);
    Query var3(num3);

    arr1.insert(arr1.end(), var3);

    Query var1(arr1);

    arr0.insert(arr0.end(), var1);

    std::vector<Query> arr4(2);

    double num5(1);
    Query var5(num5);

    arr4.insert(arr4.end(), var5);

    double num6(1);
    Query var6(num6);

    arr4.insert(arr4.end(), var6);

    Query var4(arr4);

    arr0.insert(arr0.end(), var4);

    Query var0(arr0);
  }

  SECTION("test13") {
    std::vector<Query> arr0(4);

    std::vector<Query> arr1(2);

    double num2(0);
    Query var2(num2);

    arr1.insert(arr1.end(), var2);

    double num3(1);
    Query var3(num3);

    arr1.insert(arr1.end(), var3);

    Query var1(arr1);

    arr0.insert(arr0.end(), var1);

    std::vector<Query> arr4(2);

    double num5(1);
    Query var5(num5);

    arr4.insert(arr4.end(), var5);

    double num6(2);
    Query var6(num6);

    arr4.insert(arr4.end(), var6);

    Query var4(arr4);

    arr0.insert(arr0.end(), var4);

    std::vector<Query> arr7(2);

    double num8(2);
    Query var8(num8);

    arr7.insert(arr7.end(), var8);

    double num9(3);
    Query var9(num9);

    arr7.insert(arr7.end(), var9);

    Query var7(arr7);

    arr0.insert(arr0.end(), var7);

    std::vector<Query> arr10(2);

    double num11(3);
    Query var11(num11);

    arr10.insert(arr10.end(), var11);

    double num12(4);
    Query var12(num12);

    arr10.insert(arr10.end(), var12);

    Query var10(arr10);

    arr0.insert(arr0.end(), var10);

    Query var0(arr0);
  }

  SECTION("test14") {
    std::vector<Query> arr0(3);

    std::vector<Query> arr1(2);

    double num2(0);
    Query var2(num2);

    arr1.insert(arr1.end(), var2);

    double num3(0);
    Query var3(num3);

    arr1.insert(arr1.end(), var3);

    Query var1(arr1);

    arr0.insert(arr0.end(), var1);

    std::vector<Query> arr4(2);

    double num5(1);
    Query var5(num5);

    arr4.insert(arr4.end(), var5);

    double num6(1);
    Query var6(num6);

    arr4.insert(arr4.end(), var6);

    Query var4(arr4);

    arr0.insert(arr0.end(), var4);

    std::vector<Query> arr7(2);

    double num8(2);
    Query var8(num8);

    arr7.insert(arr7.end(), var8);

    double num9(2);
    Query var9(num9);

    arr7.insert(arr7.end(), var9);

    Query var7(arr7);

    arr0.insert(arr0.end(), var7);

    Query var0(arr0);
  }

  SECTION("test15") {
    std::map<std::string, Query> map0;

    std::string src1("func", 4);
    Query var1(src1);

    std::string src2("err", 3);
    Query var2(src2);

    map0.insert({src1, var2});

    Query var0(map0);
  }

  SECTION("test16") {
    std::map<std::string, Query> map0;

    std::string src1("func", 4);
    Query var1(src1);

    std::string src2("err", 3);
    Query var2(src2);

    map0.insert({src1, var2});

    Query var0(map0);
  }

  SECTION("test17") {
    std::vector<Query> arr0(3);

    std::vector<Query> arr1(1);

    double num2(0);
    Query var2(num2);

    arr1.insert(arr1.end(), var2);

    Query var1(arr1);

    arr0.insert(arr0.end(), var1);

    std::vector<Query> arr3(1);

    double num4(1);
    Query var4(num4);

    arr3.insert(arr3.end(), var4);

    Query var3(arr3);

    arr0.insert(arr0.end(), var3);

    std::vector<Query> arr5(1);

    double num6(2);
    Query var6(num6);

    arr5.insert(arr5.end(), var6);

    Query var5(arr5);

    arr0.insert(arr0.end(), var5);

    Query var0(arr0);
  }

  SECTION("test18") {
    std::vector<Query> arr0(3);

    double num1(1);
    Query var1(num1);

    arr0.insert(arr0.end(), var1);

    double num2(2);
    Query var2(num2);

    arr0.insert(arr0.end(), var2);

    double num3(3);
    Query var3(num3);

    arr0.insert(arr0.end(), var3);

    Query var0(arr0);
  }

  SECTION("test19") {
    std::vector<Query> arr0(3);

    std::vector<Query> arr1(2);

    double num2(0);
    Query var2(num2);

    arr1.insert(arr1.end(), var2);

    double num3(0);
    Query var3(num3);

    arr1.insert(arr1.end(), var3);

    Query var1(arr1);

    arr0.insert(arr0.end(), var1);

    std::vector<Query> arr4(2);

    double num5(1);
    Query var5(num5);

    arr4.insert(arr4.end(), var5);

    double num6(1);
    Query var6(num6);

    arr4.insert(arr4.end(), var6);

    Query var4(arr4);

    arr0.insert(arr0.end(), var4);

    std::vector<Query> arr7(2);

    double num8(2);
    Query var8(num8);

    arr7.insert(arr7.end(), var8);

    double num9(2);
    Query var9(num9);

    arr7.insert(arr7.end(), var9);

    Query var7(arr7);

    arr0.insert(arr0.end(), var7);

    Query var0(arr0);
  }
}

File: libReQLTests/polyglot/transform/map.m
// Copyright 2015 Adam Grandquist

#import <Cocoa/Cocoa.h>
#import <XCTest/XCTest.h>
#import <libReQL.h>

@interface ObjcTestsTheRqlMapFunction : XCTestCase

@end

@implementation ObjcTestsTheRqlMapFunction


- (void)test0 {
  ReQLConnection *connection = [ReQLConnection new];
  XCTAssert([connection isOpen]);
  NSString *var0 = @"STREAM";
  XCTAssert([var0 isEqualTo:var0]);
  [connection close];
}

- (void)test1 {
  ReQLConnection *connection = [ReQLConnection new];
  XCTAssert([connection isOpen]);
  NSString *var0 = @"STREAM";
  XCTAssert([var0 isEqualTo:var0]);
  [connection close];
}

- (void)test2 {
  ReQLConnection *connection = [ReQLConnection new];
  XCTAssert([connection isOpen]);
  NSString *var0 = @"ARRAY";
  XCTAssert([var0 isEqualTo:var0]);
  [connection close];
}

- (void)test3 {
  ReQLConnection *connection = [ReQLConnection new];
  XCTAssert([connection isOpen]);
  NSMutableArray *var0 = [NSMutableArray arrayWithCapacity:3];
  NSNumber *var1 = @0;
  [var0 addObject:var1];
  NSNumber *var2 = @0;
  [var0 addObject:var2];
  NSNumber *var3 = @0;
  [var0 addObject:var3];
  XCTAssert([var0 isEqualTo:var0]);
  [connection close];
}

- (void)test4 {
  ReQLConnection *connection = [ReQLConnection new];
  XCTAssert([connection isOpen]);
  NSMutableArray *var0 = [NSMutableArray arrayWithCapacity:3];
  NSNumber *var1 = @0;
  [var0 addObject:var1];
  NSNumber *var2 = @0;
  [var0 addObject:var2];
  NSNumber *var3 = @0;
  [var0 addObject:var3];
  XCTAssert([var0 isEqualTo:var0]);
  [connection close];
}

- (void)test5 {
  ReQLConnection *connection = [ReQLConnection new];
  XCTAssert([connection isOpen]);
  NSMutableArray *var0 = [NSMutableArray arrayWithCapacity:1];
  NSMutableArray *var1 = [NSMutableArray arrayWithCapacity:1];
  NSNumber *var2 = @1;
  [var1 addObject:var2];
  [var0 addObject:var1];
  XCTAssert([var0 isEqualTo:var0]);
  [connection close];
}

- (void)test6 {
  ReQLConnection *connection = [ReQLConnection new];
  XCTAssert([connection isOpen]);
  NSMutableArray *var0 = [NSMutableArray arrayWithCapacity:1];
  NSMutableArray *var1 = [NSMutableArray arrayWithCapacity:2];
  NSNumber *var2 = @1;
  [var1 addObject:var2];
  NSNumber *var3 = @1;
  [var1 addObject:var3];
  [var0 addObject:var1];
  XCTAssert([var0 isEqualTo:var0]);
  [connection close];
}

- (void)test7 {
  ReQLConnection *connection = [ReQLConnection new];
  XCTAssert([connection isOpen]);
  NSMutableArray *var0 = [NSMutableArray arrayWithCapacity:1];
  NSMutableArray *var1 = [NSMutableArray arrayWithCapacity:3];
  NSNumber *var2 = @1;
  [var1 addObject:var2];
  NSNumber *var3 = @1;
  [var1 addObject:var3];
  NSNumber *var4 = @1;
  [var1 addObject:var4];
  [var0 addObject:var1];
  XCTAssert([var0 isEqualTo:var0]);
  [connection close];
}

- (void)test8 {
  ReQLConnection *connection = [ReQLConnection new];
  XCTAssert([connection isOpen]);
  NSMutableDictionary *var0 = [NSMutableDictionary dictionaryWithCapacity:1];
  NSString *var1 = @"func";
  NSString *var2 = @"err";
  [var0 setObject:var2 forKey:var1];
  XCTAssert([var0 isEqualTo:var0]);
  [connection close];
}

- (void)test9 {
  ReQLConnection *connection = [ReQLConnection new];
  XCTAssert([connection isOpen]);
  NSMutableDictionary *var0 = [NSMutableDictionary dictionaryWithCapacity:1];
  NSString *var1 = @"func";
  NSString *var2 = @"err";
  [var0 setObject:var2 forKey:var1];
  XCTAssert([var0 isEqualTo:var0]);
  [connection close];
}

- (void)test10 {
  ReQLConnection *connection = [ReQLConnection new];
  XCTAssert([connection isOpen]);
  NSArray *var0 = [NSArray array];
  XCTAssert([var0 isEqualTo:var0]);
  [connection close];
}

- (void)test11 {
  ReQLConnection *connection = [ReQLConnection new];
  XCTAssert([connection isOpen]);
  NSMutableArray *var0 = [NSMutableArray arrayWithCapacity:2];
  NSMutableArray *var1 = [NSMutableArray arrayWithCapacity:2];
  NSNumber *var2 = @1;
  [var1 addObject:var2];
  NSNumber *var3 = @1;
  [var1 addObject:var3];
  [var0 addObject:var1];
  NSMutableArray *var4 = [NSMutableArray arrayWithCapacity:2];
  NSNumber *var5 = @2;
  [var4 addObject:var5];
  NSNumber *var6 = @2;
  [var4 addObject:var6];
  [var0 addObject:var4];
  XCTAssert([var0 isEqualTo:var0]);
  [connection close];
}

- (void)test12 {
  ReQLConnection *connection = [ReQLConnection new];
  XCTAssert([connection isOpen]);
  NSMutableArray *var0 = [NSMutableArray arrayWithCapacity:2];
  NSMutableArray *var1 = [NSMutableArray arrayWithCapacity:2];
  NSNumber *var2 = @0;
  [var1 addObject:var2];
  NSNumber *var3 = @0;
  [var1 addObject:var3];
  [var0 addObject:var1];
  NSMutableArray *var4 = [NSMutableArray arrayWithCapacity:2];
  NSNumber *var5 = @1;
  [var4 addObject:var5];
  NSNumber *var6 = @1;
  [var4 addObject:var6];
  [var0 addObject:var4];
  XCTAssert([var0 isEqualTo:var0]);
  [connection close];
}

- (void)test13 {
  ReQLConnection *connection = [ReQLConnection new];
  XCTAssert([connection isOpen]);
  NSMutableArray *var0 = [NSMutableArray arrayWithCapacity:4];
  NSMutableArray *var1 = [NSMutableArray arrayWithCapacity:2];
  NSNumber *var2 = @0;
  [var1 addObject:var2];
  NSNumber *var3 = @1;
  [var1 addObject:var3];
  [var0 addObject:var1];
  NSMutableArray *var4 = [NSMutableArray arrayWithCapacity:2];
  NSNumber *var5 = @1;
  [var4 addObject:var5];
  NSNumber *var6 = @2;
  [var4 addObject:var6];
  [var0 addObject:var4];
  NSMutableArray *var7 = [NSMutableArray arrayWithCapacity:2];
  NSNumber *var8 = @2;
  [var7 addObject:var8];
  NSNumber *var9 = @3;
  [var7 addObject:var9];
  [var0 addObject:var7];
  NSMutableArray *var10 = [NSMutableArray arrayWithCapacity:2];
  NSNumber *var11 = @3;
  [var10 addObject:var11];
  NSNumber *var12 = @4;
  [var10 addObject:var12];
  [var0 addObject:var10];
  XCTAssert([var0 isEqualTo:var0]);
  [connection close];
}

- (void)test14 {
  ReQLConnection *connection = [ReQLConnection new];
  XCTAssert([connection isOpen]);
  NSMutableArray *var0 = [NSMutableArray arrayWithCapacity:3];
  NSMutableArray *var1 = [NSMutableArray arrayWithCapacity:2];
  NSNumber *var2 = @0;
  [var1 addObject:var2];
  NSNumber *var3 = @0;
  [var1 addObject:var3];
  [var0 addObject:var1];
  NSMutableArray *var4 = [NSMutableArray arrayWithCapacity:2];
  NSNumber *var5 = @1;
  [var4 addObject:var5];
  NSNumber *var6 = @1;
  [var4 addObject:var6];
  [var0 addObject:var4];
  NSMutableArray *var7 = [NSMutableArray arrayWithCapacity:2];
  NSNumber *var8 = @2;
  [var7 addObject:var8];
  NSNumber *var9 = @2;
  [var7 addObject:var9];
  [var0 addObject:var7];
  XCTAssert([var0 isEqualTo:var0]);
  [connection close];
}

- (void)test15 {
  ReQLConnection *connection = [ReQLConnection new];
  XCTAssert([connection isOpen]);
  NSMutableDictionary *var0 = [NSMutableDictionary dictionaryWithCapacity:1];
  NSString *var1 = @"func";
  NSString *var2 = @"err";
  [var0 setObject:var2 forKey:var1];
  XCTAssert([var0 isEqualTo:var0]);
  [connection close];
}

- (void)test16 {
  ReQLConnection *connection = [ReQLConnection new];
  XCTAssert([connection isOpen]);
  NSMutableDictionary *var0 = [NSMutableDictionary dictionaryWithCapacity:1];
  NSString *var1 = @"func";
  NSString *var2 = @"err";
  [var0 setObject:var2 forKey:var1];
  XCTAssert([var0 isEqualTo:var0]);
  [connection close];
}

- (void)test17 {
  ReQLConnection *connection = [ReQLConnection new];
  XCTAssert([connection isOpen]);
  NSMutableArray *var0 = [NSMutableArray arrayWithCapacity:3];
  NSMutableArray *var1 = [NSMutableArray arrayWithCapacity:1];
  NSNumber *var2 = @0;
  [var1 addObject:var2];
  [var0 addObject:var1];
  NSMutableArray *var3 = [NSMutableArray arrayWithCapacity:1];
  NSNumber *var4 = @1;
  [var3 addObject:var4];
  [var0 addObject:var3];
  NSMutableArray *var5 = [NSMutableArray arrayWithCapacity:1];
  NSNumber *var6 = @2;
  [var5 addObject:var6];
  [var0 addObject:var5];
  XCTAssert([var0 isEqualTo:var0]);
  [connection close];
}

- (void)test18 {
  ReQLConnection *connection = [ReQLConnection new];
  XCTAssert([connection isOpen]);
  NSMutableArray *var0 = [NSMutableArray arrayWithCapacity:3];
  NSNumber *var1 = @1;
  [var0 addObject:var1];
  NSNumber *var2 = @2;
  [var0 addObject:var2];
  NSNumber *var3 = @3;
  [var0 addObject:var3];
  XCTAssert([var0 isEqualTo:var0]);
  [connection close];
}

- (void)test19 {
  ReQLConnection *connection = [ReQLConnection new];
  XCTAssert([connection isOpen]);
  NSMutableArray *var0 = [NSMutableArray arrayWithCapacity:3];
  NSMutableArray *var1 = [NSMutableArray arrayWithCapacity:2];
  NSNumber *var2 = @0;
  [var1 addObject:var2];
  NSNumber *var3 = @0;
  [var1 addObject:var3];
  [var0 addObject:var1];
  NSMutableArray *var4 = [NSMutableArray arrayWithCapacity:2];
  NSNumber *var5 = @1;
  [var4 addObject:var5];
  NSNumber *var6 = @1;
  [var4 addObject:var6];
  [var0 addObject:var4];
  NSMutableArray *var7 = [NSMutableArray arrayWithCapacity:2];
  NSNumber *var8 = @2;
  [var7 addObject:var8];
  NSNumber *var9 = @2;
  [var7 addObject:var9];
  [var0 addObject:var7];
  XCTAssert([var0 isEqualTo:var0]);
  [connection close];
}

@end
